src/editor/model.ts

```diff
@@ -38,6 +65,11 @@ interface IEditorModel extends IDisposable {
   edgeRequested: ISignal<IEditorModel, EdgeLocation>;
 
   /**
+   * A signal emitted when a keydown event occurs in the editor.
+   */
+  keydown: ISignal<IEditorModel, IEditorKeydown>;
```

Comment:
I see this as analogous to the `edgeRequested` signal, which also goes through the model. The editor model emits those signals and the console model listens to them. That way as the console model creates new prompts, it can easily disconnect the old ones and listen to the new ones. If I move it to the widget, then the listening widget has to do more work that I think is better suited to happening in the model.